csrc/ops/alias.cpp

```diff
@@ -421,6 +421,19 @@ TensorView* transpose(TensorView* x) {
   return transpose(x, 0, 1);
 }
 
+bool hasSimilarDtype(DataType base, DataType dt) {
+  if (isComplexType(base)) {
+    return isComplexType(dt);
+  } else if (isFloatingPointType(base)) {
+    return isFloatingPointType(dt);
+  } else if (isBooleanType(base)) {
+    return isBooleanType(dt);
+  } else if (isIntegralType(base)) {
+    return isIntegralType(dt);
+  }
```

Comment:
should we throw a warning or error if encounter some types that we don't recognize?
Otherwise, this might return a negative false and that's not right.